### PR TITLE
Add optimized version of DijkstraAlgorithm

### PR DESCRIPTION
**Pull request by likespro**
Speed up Dijkstra Algorithm by using another Dijkstra variant that is generally faster.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
Could you add this as a separate optimized implementation? Both are good. The proposed one is faster, but the original one is simpler and easier to understand.

File: src/main/java/com/thealgorithms/datastructures/graphs/DijkstraOptimizedAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        printDistances(distances);
+        return distances;
+    }
+
+    /**
+     * Prints the shortest distances from the source vertex to all other vertices.
+     *
+     * @param distances The array of shortest distances.
+     */
+    private void printDistances(int[] distances) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/graphs/DijkstraOptimizedAlgorithm.java:73]: Please remove printing

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/datastructures/graphs/DijkstraOptimizedAlgorithm.java (hunk 1 above)
Comment (likespro) [reply src/main/java/com/thealgorithms/datastructures/graphs/DijkstraOptimizedAlgorithm.java]: Removed

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**